src/runtime/server/middleware/session/index.ts

```diff
@@ -20,9 +21,62 @@ const safeSetCookie = (event: H3Event, name: string, value: string) => setCookie
 export declare interface Session {
   id: string
   createdAt: Date
+  ip?: string
   [key: string]: any
 }
 
+const argon2Options = {
```

Comment:
Nit: I would move all newly-added logic into a separate file, eg `ip-pinning.ts`

Reply:
Some of it will have to be "intrusive", but I can move the implementation to their dedicated file for sure

Reply:
What do you mean by "intrusive"? The motivation behind the suggestion is to keep your logic encapsulated in one file, so it's easily readable and next extension doesn't blow up index file even more, but since it's a "nit" suggestion it can be done at a later stage as a refactoring.